### ISSUE TITLE
How to include Negative samples/images for training SSD

### ISSUE THREAD

**Issue by navneet1083** [intel/caffe#133]
How to add negative samples while training. If i have my own datasets for 2 classes (e.g laptop and tvmonitor) and i have to include negative images/samples (which are not laptop and tvmonitor) then how should i add those images in configuration or in labels config.

Reason being to put this forward is after trained fully my network recognises window screen (with cover and some dim sunlight) as tvmonitor so i need to feed some negative samples to overcome this issue.

Or kindly suggest me some alternative to overcome this issue.

**Comment by jgong5** [comment]
How about adding a new class for your negative samples?

**Comment by navneet1083** [comment]
@jgong5 - yes i can but this specific class will not converge which could impact for overall accuracy (as it may include NON-classes categories like non-cat;non-laptop etc). Also number of examples to put in this category (negative-class) will be an issue. 
As it is the bounding-boxes which evaluate the prediction of having such objects then creating bonding-boxes will be an issue Or are you suggesting to put random (more than one) bbox in negative class samples/images ?

**Comment by ftian1** [comment]
@navneet1083 are you sure adding negative samples is a good idea? I think your concern is very generic and I don't see any other typologies take such idea.

How much mAP you got after fine tuned on your own datasets? is it too low? how many training samples are there in your datasets?

**Comment by navneet1083** [comment]
@ftian1 Since i am trying with 2 classes, first gathered 300 odd samples each. mAP not crossing beyond 60. Also, i have applied only basic augmentation; planning to go ahead with some CV-related augmentation like intensity, gaussian blur, rotation etc. But before that i wanted to know the approach is good enough to proceed further or not. 
Or is there any research paper/blog you might have come across and 'd like to share?

**Comment by ftian1** [comment]
@navneet1083 
you can have a look at 
[https://blog.keras.io/building-powerful-image-classification-models-using-very-little-data.html](url)

**Comment by navneet1083** [comment]
@ftian1 getting 404-page not found

**Comment by navneet1083** [comment]
@ftian1 Oops copy-paste did work !!

**Comment by hshen14** [comment]
Close it due to no activities in recent months. Please feel free to reopen it.
